CMakeLists.txt

```diff
@@ -680,13 +519,19 @@ set(scylla_gen_sources
     "${scylla_thrift_gen_cassandra_files}"
     "${scylla_ragel_gen_protocol_parser_file}"
     "${swagger_gen_files}"
-    "${idl_gen_files}"
-    "${antlr3_gen_files}")
+    "${idl_gen_files}")
+
+add_subdirectory(alternator)
+add_subdirectory(cql3)
 
 add_executable(scylla
     ${scylla_sources}
     ${scylla_gen_sources})
 
+target_link_libraries(scylla PRIVATE
+    alternator
+    cql3)
```

Comment:
Will these be the bases of the future modules?

Reply:
@denesb exactly. see also https://github.com/tchaikov/seastar/blob/cxx-modules-impl/modules/CMakeLists.txt . it would be much simpler when porting this to C++20 modules. as CMake insists to group C++ modules using `BASE_DIR` .